flink-runtime/src/main/java/org/apache/flink/runtime/executiongraph/IntermediateResultPartition.java

```diff
@@ -30,30 +30,26 @@
 
     private final ExecutionVertex producer;
 
-    private final int partitionNumber;
-
     private final IntermediateResultPartitionID partitionId;
 
-    private List<List<ExecutionEdge>> consumers;
-
     /** Whether this partition has produced some data. */
     private boolean hasDataProduced = false;
 
     public IntermediateResultPartition(
             IntermediateResult totalResult, ExecutionVertex producer, int partitionNumber) {
         this.totalResult = totalResult;
         this.producer = producer;
-        this.partitionNumber = partitionNumber;
-        this.consumers = new ArrayList<List<ExecutionEdge>>(0);
         this.partitionId = new IntermediateResultPartitionID(totalResult.getId(), partitionNumber);
+
+        producer.getExecutionGraph().registerResultPartition(partitionId, this);
```

Comment:
I am not a huge fan of coupling components by these kind of constructs. Couldn't we register the `IntermediateResultPartition` where it is created (e.g. in the `ExecutionVertex`)?